samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -1218,11 +1224,11 @@ static void broadcast_scan_recv(const struct bt_le_scan_recv_info *info, struct
 		 */
 		if (req_recv_state == NULL && strlen(CONFIG_TARGET_BROADCAST_NAME) > 0U) {
 			struct net_buf_simple buf_copy;
-			char name[NAME_LEN] = {0};
 
 			net_buf_simple_clone(ad, &buf_copy);
 			bt_data_parse(&buf_copy, data_cb, name);
-			if (!(is_substring(CONFIG_TARGET_BROADCAST_NAME, name))) {
+
+			if (!device_found) {
```

Comment:
I think you are missing a declaration of `device_found` and passing it on the `data_cb` by modifying `bt_data_parse(&buf_copy, data_cb, name);` to be `bt_data_parse(&buf_copy, data_cb, &device_found);`

Reply:
Sorry, was a bit quick on the push-button. Fixed